src/quickgui/qgsquickutils.cpp

```diff
@@ -0,0 +1,77 @@
+/***************************************************************************
+  qgsquickutils.cpp
+  --------------------------------------
+  Date                 : Nov 2017
+  Copyright            : (C) 2017 by Peter Petrik
+  Email                : zilolv at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include <QDesktopWidget>
+#include <QDir>
+#include <QFile>
+#include <QFileInfo>
+#include <QString>
+#include <QThread>
+
+#include "qgis.h"
+#include "qgscoordinatereferencesystem.h"
+#include "qgscoordinatetransform.h"
+#include "qgsdistancearea.h"
+#include "qgslogger.h"
+#include "qgsvectorlayer.h"
+
+#include "qgsquickmapsettings.h"
+#include "qgsquickutils.h"
+#include "qgsunittypes.h"
+
+QgsQuickUtils *QgsQuickUtils::sInstance = nullptr;
+
+QgsQuickUtils *QgsQuickUtils::instance()
+{
+  if ( !sInstance )
+  {
+    QgsDebugMsg( QStringLiteral( "QgsQuickUtils created: %1" ).arg( long( QThread::currentThreadId() ) ) );
+    sInstance = new QgsQuickUtils();
```

Comment:
I do not think so: https://github.com/qgis/QGIS/pull/6842/files/ca3243d8656e0431b2a0e00803db717e580d22d6#diff-09ffc66fca3d9120f66c33a81f485949R41

Reply:
Is this the standard way to deal with singletons in QML context? I'd expect the `new` to happen where it is registered. With this approach here responsibilities for creation and cleanup are distributed in a non-intuitive way.

Reply:
Actually, we needed to set dp from C++ code at some point, but later we decided that it can be done automatically. So singleton in no longer needed in C++ context, so I implemented [QML singleton](http://doc.qt.io/qt-5/qqmlengine.html#qmlRegisterSingletonType) as suggested in QT docs.